saleor/payment/gateways/np_atobarai/__init__.py

```diff
@@ -0,0 +1,160 @@
+import logging
+import os
+from typing import List
+
+from ....order.models import Fulfillment
+from ... import PaymentError, TransactionKind
+from ...interface import GatewayResponse, PaymentData
+from ...models import Payment
+from . import api
+from .api_types import ApiConfig, PaymentStatus
+from .const import NP_PLUGIN_ID
+from .utils import (
+    get_fulfillment_for_order,
+    get_payment_name,
+    get_shipping_company_code,
+    notify_dashboard,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def parse_errors(errors: List[str]) -> str:
+    # Field error of Transaction db model has max length of 256
+    # Error codes have max length of 11
+    # We are limiting errors to maximum of 11 codes, because:
+    # 11 * 11 + 10 * 2 (max length of linesep) == 141 < 256
+    return os.linesep.join(errors[:11])
+
+
+def process_payment(
+    payment_information: PaymentData, config: ApiConfig
+) -> GatewayResponse:
+    """Create new transaction in NP Atobarai.
+
+    Returns unsuccessful response if payment status from
+    NP response is PENDING or FAILED.
+
+    :raises PaymentError: When the specified payment does not exist.
+    """
+    payment_id = payment_information.payment_id
+    payment = Payment.objects.filter(pk=payment_id).first()
+
+    if not payment:
+        logger.error(
+            "Payment with id %s does not exist",
+            payment_information.graphql_payment_id,
+        )
+        raise PaymentError("Payment does not exist.")
+
+    result = api.register_transaction(payment.order, config, payment_information)
+
+    return GatewayResponse(
+        is_success=result.status == PaymentStatus.SUCCESS,
+        action_required=False,
+        kind=TransactionKind.CAPTURE,
+        amount=payment_information.amount,
+        currency=payment_information.currency,
+        transaction_id=result.psp_reference,
+        error=parse_errors(result.errors),
+        raw_response=result.raw_response,
+        psp_reference=result.psp_reference,
+    )
+
+
+def capture(payment_information: PaymentData, config: ApiConfig) -> GatewayResponse:
+    raise NotImplementedError
+
+
+def void(payment_information: PaymentData, config: ApiConfig) -> GatewayResponse:
+    raise NotImplementedError
+
+
+def tracking_number_updated(fulfillment: Fulfillment, config: ApiConfig) -> None:
+    """Event callback on updating order tracking number.
+
+    Captures payment in NP Atobarai and sends dashboard notification of the result.
+    """
+    order = fulfillment.order
+    payments = order.payments.filter(gateway=NP_PLUGIN_ID, is_active=True)
+
+    if payments:
+        results = [
+            api.report_fulfillment(config, payment, fulfillment) for payment in payments
+        ]
+    else:
+        results = [("", ["No active payments for this order"], False)]
```

Comment:
Consider just 
```
notify_dashboard(order, "No active payments for this order")
return
```
Instead of force creating data for `loop` below. Notice that you produce `""`  payment_id so in line 98/99 the error msg will be lack of payment name information.

Reply:
Resolved in recent commi